### PR TITLE
restyle API Ref

### PR DESCRIPTION
**Pull request by piskvorky**
Render a nicer API reference page. Continues from #1809.

I had trouble merging `develop` into this PR -- there were files with trailing whitespace (`*_inner.c`), and even two files with Windows line endings: `lsi_dispatcher.py`, `lsi_worker.py`, which prevented me from committing the resolved conflicts.

I solved the conflicts by removing trailing whitespace, converting line endings to Unix and then finishing the merge commit.

I also fixed the content of the documentation of the distributed LSI and LDA files, since it used incorrect docstring style and introduced some minor documentation bugs, in PR #1912 (specifically commit 1611f3a477686bf9f462d0ed09b1daff2a58f09e).

### CODE REVIEW

**Comment by piskvorky** [comment]
@anotherbugmaster I'm getting some CircleCI errors after the merge. Can you please have a look?

**Comment by anotherbugmaster** [comment]
@piskvorky, I'm on a vacation til 5th of July, but I'll have a look tonight

**Comment by anotherbugmaster** [comment]
image

Fixed.

Removed *.rst with inner/outer d2v and w2v for now (not sure where to place them in the new API reference).

**Comment by piskvorky** [comment]
Strange, I see some more merge conflicts. And CircleCI failing. @menshikh-iv do you know what's going on?

**Comment by piskvorky** [comment]
I resolved the merge conflict. It was the recently merged non-API PR, which changed some files that this PR deleted.

We cannot close this PR, it contains multiple important fixes. I suggest closing the other PR (I see some regressions in the recent commits there).

File: gensim/models/tfidfmodel.py [hunk 1]
```diff
@@ -301,10 +301,12 @@ def __init__(self, corpus=None, id2word=None, dictionary=None, wlocal=utils.iden
             Only when `pivot` is not None will pivoted document length normalization be applied.
             Otherwise, regular TfIdf is used.
         slope : float, optional
-            Parameter required by pivoted document length normalization which determines the slope to which
-            the `old normalization` can be tilted. This parameter only works when `pivot` is defined.
+            It is the parameter required by pivoted document length normalization which determines the slope to which
```
Comment (piskvorky) [inline gensim/models/tfidfmodel.py]: Regression (PR contained had a newer, fixed version).

Out of curiosity -- what was this conflict from? This branch was already up-to-date with `develop`.

**Comment by menshikh-iv** [comment]
This PR is important, but we need to fix current issues first, CC @anotherbugmaster 
1. "old apiref" doesn't looks like current apiref (3.4.0), that's a bug
2. Many useful function & classes doesn't mention in new apiref
3. General "linkage" issue (before we have one rst for module, now - rst for class/function, but not for module -> missing great module docstrings + general layout, this approach is OK for numpy, but bad for Gensim).

**Comment by piskvorky** [comment]
Apparently this PR contains some doc fixes, now lost in a sea of weird merges and abandoned code. I no longer recall what all this was about.

Is it worth fishing out those fixes and applying them to the current `develop`?

**Pull request closed without merging**